recipes/generate.py

```diff
@@ -81,9 +81,40 @@ def _setup_model(
 
         return model
 
+    def convert_prompt_to_tokens(
+        self,
+        prompt: Union[DictConfig, str],
+        chat_format: ChatFormat,
+        instruct_template: InstructTemplate,
+    ) -> List[Message]:
+        # Should only be chat-style prompt or instruct-style prompt
+        if chat_format and instruct_template:
+            raise ValueError(
+                "Cannot pass both chat format and instruct template for generation"
+            )
+
+        # A single string can be formatted with instruct_template
+        if isinstance(prompt, str):
```

Comment:
this highlights a reason why I want to move towards combining instruct_template / chat_format. You can just treat a single string as a user message, then apply instruct template only on the user message. I don't like the logic here but making this cleaner might require updating InstructTemplate / ChatFormat

Reply:
Yeah this is def a short-term fix. I am not getting into instruct template/chat format here cause that is a heavier refactor